iox_messaging.proto

```diff
@@ -0,0 +1,352 @@
+/*****************************************************************************
+ *
+ * Copyright (C) 2023, Geotab Inc.
+ *
+ ******************************************************************************
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *****************************************************************************/
+
+/******************************************************************************
+Revision  Date        Author  Notes
+0.1       2023-04-05  YL      Initial draft
+0.2       2023-04-24  YL      Fix gps_time field number, rename Publish fields
+0.3       2023-05-03  YL      Add clear_subs_ack
+0.4       2023-07-14  NT      Add comments
+*****************************************************************************/
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+/**
+ * This Proto file defines message payloads for the Third party Message Type 0x8C (Protobuf Data Packet from IOX to GO) and Message Type 0x26 (Protobuf Data Packet from GO to IOX).
+ * The protobufs defined here follows a simple pub/sub protocol, where a third party IOX device can subscribe to one of the supported TOPICs (enum Topic) 
+ * and receive the desired information from the GO.
+ * The appropriate master switch needs to be set to use the pub/sub functionality correctly.
```

Comment:
Consider using another term for 'master switch' as it is likely meaningless for external integrators.

Reply:
Will replace all 'master switch' with 'feature flag'.